### PR TITLE
Added 2 fields under WITH option.

### LINKED ISSUE #367: Document new parameters/options for Kinesis
### Related code PR

https://github.com/risingwavelabs/risingwave/pull/6317

### Which part(s) of the docs might be affected or should be updated? And how?

Create source - Kinesis - parameters section. Please see the test cases in the code PR for details. 

### Reference

_No response_

### PR DESCRIPTION
**Pull request by emile-00**
Added 2 fields under WITH option.

## Info
- **Description**: 
Documented 2 new fields under WITH option for Kinesis source.

- **Related code PR**: 
https://github.com/risingwavelabs/risingwave/pull/6317

- **Related doc issue**: 
Resolves https://github.com/risingwavelabs/risingwave-docs/issues/367

- **Notes**: 
[ Any additional information? ]

## Before merging
  - [ ] (For version-specific PR) I have selected the corresponding software version in **Milestone** and linked the related doc issue to this PR in **Development**.
  - [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`bernscode`, `CharlieSYH`, `emile-00`, & `hengm3467`). 
  - [ ] I have checked the doc site preview, and the updated parts look good. <details><summary>How?</summary>Scroll down and open this link: image
</details>

## Published doc pages
  [ After merging this PR, edit the description to include the links to the updated doc pages here. For example, https://www.risingwave.dev/docs/latest/intro/. ]

### CODE REVIEW

File: docs/sql/create-source/create-source-kinesis.md [hunk 1]
```diff
@@ -31,6 +31,8 @@ ROW FORMAT AVRO | JSON | PROTOBUF MESSAGE 'main_message';
 |aws.credentials.session_token	|None	|String	|The session token associated with the credentials. Temporary Session Credentials.	|False|
 |aws.credentials.role.arn	|None	|String |The Amazon Resource Name (ARN) of the role to assume.		|False|
 |aws.credentials.role.external_id	|None	|String	|The [external id](https://aws.amazon.com/blogs/security/how-to-use-external-id-when-granting-access-to-your-aws-resources/) used to authorize access to third-party resources.	|False|
+|scan.startup.mode |earliest  |String| The Kinesis consumer starts consuming data from the commit offset. This includes three values: 'earliest', 'latest', and 'sequence_number'.|False|
```
Comment (hengm3467) [inline docs/sql/create-source/create-source-kinesis.md]: Maybe rephrase to something like: The startup mode for Kinesis consumer. Supported modes: `earliest` (starting from the earliest offset), `latest` (starting from the latest offset), and `sequence_number` (starting from a specific sequence number, which needs to be configured by `scan.startup.sequence_number`).

File: docs/sql/create-source/create-source-kinesis.md [hunk 2]
```diff
@@ -31,6 +31,8 @@ ROW FORMAT AVRO | JSON | PROTOBUF MESSAGE 'main_message';
...
+|scan.startup.sequence_number |None | int64| Specify the sequence number to start consuming from. | True if `scan.startup.mode` = 'sequence_number', otherwise False|
```
Comment (CharlieSYH) [inline docs/sql/create-source/create-source-kinesis.md]: Not sure if we should stick with the data type names we list in Data types.
image

**Review by CharlieSYH — APPROVED** [review]

**Review by hengm3467 — APPROVED** [review]
